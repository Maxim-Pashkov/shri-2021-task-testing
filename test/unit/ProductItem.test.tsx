import React from 'react';
import { it, expect, describe } from '@jest/globals';
import { render } from '@testing-library/react';
import { getApplication } from './getApplication';
import { ProductItem } from '../../src/client/components/ProductItem';

const product = {
    name: 'test name',
    price: 10,
    count: 1,
    id: 5,
};

describe('ProductItem test', () => {
    it('item is exists', () => {
        const { application } = getApplication(
            () => <ProductItem product={product} />,
            {}
        );

        const { container } = render(application);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('selected item is exists', () => {
        const cartState = {
            [product.id]: product,
        };

        const { application } = getApplication(
            () => <ProductItem product={product} />,
            cartState
        );

        const { container } = render(application);

        expect(container.firstChild).toMatchSnapshot();
    });
});
