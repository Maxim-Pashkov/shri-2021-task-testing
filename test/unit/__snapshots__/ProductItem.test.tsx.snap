// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem test item is exists 1`] = `
<div
  class="ProductItem card w-100 mb-4"
  data-testid="5"
>
  <img
    class="Image card-img-top"
    src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkMAYAADkANVKH3ScAAAAASUVORK5CYII="
  />
  <div
    class="card-body"
  >
    <h5
      class="ProductItem-Name card-title"
    >
      test name
    </h5>
    <p
      class="ProductItem-Price card-text"
    >
      $
      10
    </p>
    <a
      class="ProductItem-DetailsLink card-link"
      href="/catalog/5"
    >
      Details
    </a>
  </div>
</div>
`;

exports[`ProductItem test selected item is exists 1`] = `
<div
  class="ProductItem card w-100 mb-4"
  data-testid="5"
>
  <img
    class="Image card-img-top"
    src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkMAYAADkANVKH3ScAAAAASUVORK5CYII="
  />
  <div
    class="card-body"
  >
    <h5
      class="ProductItem-Name card-title"
    >
      test name
    </h5>
    <p
      class="ProductItem-Price card-text"
    >
      $
      10
    </p>
    <a
      class="ProductItem-DetailsLink card-link"
      href="/catalog/5"
    >
      Details
    </a>
    <span
      class="CartBadge text-success mx-3"
    >
      Item in cart
    </span>
  </div>
</div>
`;
